rules/no-error-ctor-with-notthrows.js

```diff
@@ -0,0 +1,50 @@
+'use strict';
+const {visitIf} = require('enhance-visitors');
+const util = require('../util');
+const createAvaRule = require('../create-ava-rule');
+
+const errorNameRegex = /^([A-Z][a-z\d]*)*Error$/;
+
+const create = context => {
+	const ava = createAvaRule();
+
+	return ava.merge({
+		CallExpression: visitIf([
+			ava.isInTestFile,
+			ava.isInTestNode
+		])(node => {
+
+			if (typeof node.callee.property === 'undefined') {
+				return;
+			}
+
+			const functionArgIndex = node.arguments.length - 1;
+
+			if (functionArgIndex !== 1) {
+				return;
+			}
+
+			const calleeProperty = node.callee.property.name;
```

Comment:
Consider also check 
i) `node.callee.type` is of type `MemberExpression`, which also implies `node.callee.property` is not `undefined`
ii) `util.getNameOfRootNodeObject(node.callee) === 't';`

The following example explains why the checks above should be practitioned
```js
const { notThrows } = require("./my-custom-not-throws");
notThrows(foo);
```

Reply:
```
valid: [
    `const { notThrows } = require("./my-custom-not-throws");
     notThrows(foo);`
]
```

Hey, thanks for reviewing the PR. But this already passes the tests for me, can you explain a bit about how it will really be needed. Sorry if I miss silly things, just learning. :)

Reply:
>  this already passes the tests for me

The test passed because `ava.isInTestFile` is false and the whole node would not be visited if one do not wrap them in an ava test node. So here are the example of the two checks I mentioned above
1) `node.callee.type` is not an `MemberExpression`
```js

const { notThrows } = require("./my-custom-not-throws")
${header}
		test('some test',t => {
			notThrows(foo);
		});`,
```
2) `util.getNameOfRootNodeObject(node.callee)` is not `t`

```js
const myCustomNotThrows = require("./my-custom-not-throws")
${header}
		test('some test',t => {
			myCustomNotThrows.notThrows(foo);
		});`,
```

Feel free to ask any questions as everyone is learning new things. It is great to explain into details when you got response.

Reply:
> The test passed because `ava.isInTestFile` is false and the whole node would not be visited if one do not wrap them in an ava test node. So here are the 

This is what confuses me if ava automatically protects us then is it important to do these checks?

Reply:
ping.

Reply:
Thanks for the ping. I had a heavy workload recently.

> This is what confuses me if ava automatically protects us then is it important to do these checks?

`ava.isInTestFile` ([source](https://github.com/avajs/eslint-plugin-ava/blob/ff4f7606134ee167d0330284d59110033b3287b9/create-ava-rule.js#L45)) and `ava.isInTestNode` ([source](https://github.com/avajs/eslint-plugin-ava/blob/ff4f7606134ee167d0330284d59110033b3287b9/create-ava-rule.js#L80))limits our eslint rules only in the following scope.

For example, let's say you want to create an ava eslint rule `no-foo-call-in-test ` which disallow the usage of `foo()` inside the test implementation.
```js
test('some test',t => {  // <-- Am I inside an ava test? Looks good, let's check.
	foo() // <-- bail!!
});
foo()
// <-- I will not check the code here since I am not in ava test node.
```

So in your case, we are checking if a `CallExpression` is `t.notThrows()`, we can not simply assume that every `CallExpression` with the name `notThrows` is a valid ava call. In the true negative example I mentioned above,
```js
const { notThrows } = require("./my-custom-not-throws")
test('some test',t => { // <-- Am I inside an ava test? Looks good, let's check.
	notThrows(foo);  // <-- Is it a CallExpression? Yes. Is identifier name `notThrows`? Yes, we bail here
});

notThrows(foo); // <-- I will not check this node since I am not in a ava test.
```
Apparently it is not `t.notThrows` call so we should mark it as valid.

Reply:
Thanks. It is done, can you double-check? @JLHwung